### PR TITLE
llama async

### PR DESCRIPTION
**Pull request by piotrm0**
Instrument llama_index query engine's `aquery` as well as chat engine's `chat`, `achat`, `stream_chat`, and `astream_chat`. The streaming versions presently produce records before the streams are complete hence only have an incomplete account.

### CODE REVIEW

File: trulens_eval/trulens_eval/tru_llama.py [hunk 1]
```diff
@@ -179,11 +265,326 @@ def query_with_record(self, str_or_query_bundle) -> Tuple[Response, Record]:
...
+
+        assert isinstance(ret, AgentChatResponse)
+
+        ret_record_args['main_output'] = ret.response
+
         ret_record = self._post_record(
             ret_record_args, error, cost, start_time, end_time, record
         )
 
         return ret, ret_record
```
Comment (rshih32) [inline trulens_eval/trulens_eval/tru_llama.py:467]: how much of this flow can we start modularizing?

File: trulens_eval/trulens_eval/util.py [hunk 2]
```diff
@@ -61,12 +63,13 @@
 
 # Optional requirements.
 
-langchain_version = "0.0.230"
-
+llama_version = "0.7.16"
 REQUIREMENT_LLAMA = (
     "llama_index 0.6.24 or above is required for instrumenting llama_index apps. "
```
Comment (joshreini1) [inline trulens_eval/trulens_eval/util.py:68]: use {llama_version} instead of hardcoded 0.6.24

File: trulens_eval/trulens_eval/tru_llama.py (hunk 1 above)
Comment (piotrm0) [reply trulens_eval/trulens_eval/tru_llama.py:467]: A lot. There is a draft PR getting into this.

Comment (piotrm0) [reply trulens_eval/trulens_eval/tru_llama.py:467]: https://github.com/truera/trulens/pull/344

**Pull request merged**